src/test/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuildListenerTest.java

```diff
@@ -65,14 +70,24 @@ public void onStarted_sets_description() throws Exception {
   }
 
   @Test
-  public void onStarted_doesnt_set_if_there_is_no_StashCause() throws Exception {
+  public void onStarted_doesnt_set_description_if_there_is_no_StashCause() throws Exception {
     when(build.getCause(eq(StashCause.class))).thenReturn(null);
 
     stashBuildListener.onStarted(build, taskListener);
 
     verify(build, never()).setDescription(anyString());
   }
 
+  @Test
+  public void onStarted_writes_to_build_log_on_exception() throws Exception {
+    when(build.getCause(eq(StashCause.class))).thenReturn(stashCause);
+    doThrow(IOException.class).when(build).setDescription(anyString());
+
+    stashBuildListener.onStarted(build, taskListener);
+
+    verify(printStream, times(1)).println("Can't update build description: java.io.IOException");
```

Comment:
minor: instead of doing this with mocks and verify I would prefer we just have a `PrintStream` backed by a `StringBuffer`. Then we can simply inspect the `StringBuffer` (or rather the resulting String) directly

Reply:
Great idea, will do!

Reply:
Done. No exactly with `StringBuffer`, but still with a buffer in memory.